homeassistant/components/sensor/recorder.py

```diff
@@ -691,53 +687,113 @@ def validate_statistics(
-                                "supported_unit": metadata_unit,
-                            },
-                        )
+                    ir.async_create_issue(
+                        hass,
+                        DOMAIN,
+                        f"units_changed_{entity_id}",
+                        data={
+                            "issue_type": "units_changed",
+                            "statistic_id": entity_id,
+                            "state_unit": state_unit,
+                            "metadata_unit": metadata_unit,
+                            "supported_unit": metadata_unit,
+                        },
+                        is_fixable=False,
+                        severity=ir.IssueSeverity.WARNING,
+                        translation_key="units_changed",
                     )
+                else:
+                    ir.async_delete_issue(hass, DOMAIN, f"units_changed_{entity_id}")
             elif state_unit not in converter.VALID_UNITS:
                 # The state unit can't be converted to the unit in metadata
                 valid_units = (unit or "<None>" for unit in converter.VALID_UNITS)
                 valid_units_str = ", ".join(sorted(valid_units))
+                ir.async_create_issue(
+                    hass,
+                    DOMAIN,
+                    f"units_changed_{entity_id}",
+                    data={
+                        "issue_type": "units_changed",
+                        "statistic_id": entity_id,
+                        "state_unit": state_unit,
+                        "metadata_unit": metadata_unit,
+                        "supported_unit": valid_units_str,
+                    },
+                    is_fixable=False,
+                    severity=ir.IssueSeverity.WARNING,
+                    translation_key="units_changed",
+                )
+            else:
+                ir.async_delete_issue(hass, DOMAIN, f"units_changed_{entity_id}")
+
+
+def update_statistics_issues(
+    hass: HomeAssistant,
+    session: Session,
+) -> None:
+    """Validate statistics."""
+    instance = get_instance(hass)
+    sensor_states = hass.states.all(DOMAIN)
+    metadatas = statistics.get_metadata_with_session(
+        instance, session, statistic_source=RECORDER_DOMAIN
+    )
+
+    run_callback_threadsafe(hass.loop, _update_issues, hass, sensor_states, metadatas)
+
+
+def validate_statistics(
+    hass: HomeAssistant,
+) -> dict[str, list[statistics.ValidationIssue]]:
+    """Validate statistics."""
+    validation_result = defaultdict(list)
+
+    sensor_states = hass.states.all(DOMAIN)
+    metadatas = statistics.get_metadata(hass, statistic_source=RECORDER_DOMAIN)
+    sensor_entity_ids = {i.entity_id for i in sensor_states}
+    sensor_statistic_ids = set(metadatas)
+    instance = get_instance(hass)
+    entity_filter = instance.entity_filter
+
+    run_callback_threadsafe(hass.loop, _update_issues, hass, sensor_states, metadatas)
```

Comment:
If this doesn't need to wait for the the callback to finish, `loop.call_soon_threadsafe` is much more efficent